src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/AbstractDocumentDiagnosticSource.cs

```diff
@@ -31,8 +31,12 @@ protected AbstractDocumentDiagnosticSource(TDocument document)
 
     public ProjectOrDocumentId GetId() => new(Document.Id);
     public Project GetProject() => Document.Project;
-    public TextDocumentIdentifier GetDocumentIdentifier()
-        => new VSTextDocumentIdentifier { ProjectContext = ProtocolConversions.ProjectToProjectContext(Document.Project), Uri = Document.GetURI() };
+    public TextDocumentIdentifier? GetDocumentIdentifier()
+        => Document.FilePath != null
```

Comment:
Did this need to check the project file path too?

Reply:
Nope - we are reporting diagnostics on the document, we don't need the project's file path